src/main/java/seedu/address/model/module/Module.java

```diff
@@ -130,7 +131,7 @@ public void addLecture(Lecture lecture) {
      * @throws DuplicateLectureException Indicates that {@code editedLecture} is the same as another existing
      *                                   lecture in the module.
      */
-    public void setLecture(ReadOnlyLecture target, Lecture editedLecture) {
+    public void setLecture(ReadOnlyLecture target, Lecture editedLecture) throws LectureNotFoundException {
```

Comment:
No need for this as `LectureNotFoundException` is a runtime exception. But yes, thanks for importing the `LectureNotFoundException`, was a bug that I missed  this out.